Source/Game/Game/Editor/EaseCurveTool.cpp

```diff
@@ -0,0 +1,402 @@
+                    ImVec2 HRight = { p1.x, HLeft.y };
+
+                    drawList->AddLine(VLTop, VLBottom, IM_COL32(255,255,255,100));
+                    drawList->AddLine(VRTop, VRBottom, IM_COL32(255,255,255,100));
+                    drawList->AddLine(HLeft, HRight, IM_COL32(255,255,255,100));
+
+                    f32 x = GetEasePoint(_timer / _timerScale, selectedEaseType, selectedEaseName, polynomialDegree);
+
+                    f32 xOffset = x * (size.x * canvasScale);
+                    f32 x0 = p0.x + (size.x * invHalfCanvasScale) + xOffset;
+                    if (useInverse && _inverse)
+                        x0 = p0.x + (size.x * (1.f - invHalfCanvasScale)) - xOffset;
+
+                    f32 y0 = p0.y + (size.y / 2.0f);
+
+                    drawList->AddCircleFilled(ImVec2(x0, y0), 8.f, IM_COL32(255,0,0,255));
+
+
+                    drawList->AddRect(p0, p1, IM_COL32(255,255,255,255));
+                    drawList->PopClipRect();
+                }
+            }
+        }
+        ImGui::End();
+    }
+
+    void EaseCurveTool::Update(f32 deltaTime)
+    {
+        _timer += deltaTime;
+        if (_timer > _timerScale)
+        {
+            ResetTimer();
+            _inverse = !_inverse;
+        }
+    }
+
+    void EaseCurveTool::ResetTimer()
+    {
+        _timer = 0.0f;
+        _currentX = 0.0f;
+        _currentY = 0.0f;
+    }
+
+    void EaseCurveTool::ResetMinMax()
+    {
+        _minY = 10.0f;
+        _maxY = 0.0f;
+    }
+
+    f32 EaseCurveTool::GetEasePoint(f32 t, i32 n0, i32 n1, i32 polynomialDegree)
+    {
+        switch (n0)
+        {
+        case 0:
+            switch(n1)
+            {
+            case 0:
+                return Spline::Interpolation::Ease::In::Sine(t);
+            case 1:
+                return Spline::Interpolation::Ease::In::Circ(t);
+            case 2:
+                return Spline::Interpolation::Ease::In::Elastic(t);
+            case 3:
+                return Spline::Interpolation::Ease::In::Expo(t);
+            case 4:
+                return Spline::Interpolation::Ease::In::Back(t);
+            case 5:
+                return Spline::Interpolation::Ease::In::Bounce(t);
+            case 6:
+                return Spline::Interpolation::Ease::In::Quadratic(t);
+            case 7:
+                return Spline::Interpolation::Ease::In::Cubic(t);
+            case 8:
+                return Spline::Interpolation::Ease::In::Quartic(t);
+            case 9:
+                return Spline::Interpolation::Ease::In::Quintic(t);
+            case 10:
+                return Spline::Interpolation::Ease::In::Sextic(t);
+            case 11:
+                return Spline::Interpolation::Ease::In::Septic(t);
+            case 12:
+                return Spline::Interpolation::Ease::In::Octic(t);
+            case 13:
+                return Spline::Interpolation::Ease::In::Polynomial(t, polynomialDegree);
+            default:
```

Comment:
Please make a new line between the second last case, and the default case (Do this in all places where this is a problem)